lib/Transformation.ts

```diff
@@ -156,25 +158,61 @@ function transformOperator(expr: Alg.OperatorExpression)
       throw new Err.UnknownOperator(expr.operator);
     }
     const op = expr.operator as C.RegularOperator;
-    const args = expr.args.map((a) => transformAlgebra(a));
+    const args = expr.args.map((a) => transformAlgebra(a, creatorConfig));
     const func = regularFunctions.get(op);
     if (!hasCorrectArity(args, func.arity)) { throw new Err.InvalidArity(args, op); }
     return new E.Operator(args, func.apply);
   }
 }
 
+function wrapSyncFunction(f: SyncExtensionFunction, name: string): SimpleApplication {
+  return (args => {
+    let res;
+    try {
+      res = f(args.map(arg => arg.toRDF()));
+    }catch (e) {
+      throw new ExtensionFunctionError(name, e);
+    }
+    return transformRDFTermUnsafe(res);
+  });
+}
+
+function wrapAsyncFunction(f: AsyncExtensionFunction, name: string): AsyncExtensionApplication {
+  return (async args => {
+    let res;
+    try {
+      res = await f(args.map(arg => arg.toRDF()));
+    }catch (e) {
+      throw new ExtensionFunctionError(name, e);
+    }
+    return transformRDFTermUnsafe(res);
+  });
+}
 // TODO: Support passing functions to override default behaviour;
-export function transformNamed(expr: Alg.NamedExpression): E.NamedExpression {
+function transformNamed(expr: Alg.NamedExpression, creatorConfig: FunctionCreatorConfig)
+  : E.NamedExpression | E.AsyncExtensionExpression | E.SyncExtensionExpression  {
   const funcName = expr.name.value;
-  if (!C.NamedOperators.contains(funcName as C.NamedOperator)) {
-    throw new Err.UnknownNamedOperator(expr.name.value);
+  const args = expr.args.map((a) => transformAlgebra(a, creatorConfig));
+  if (C.NamedOperators.contains(funcName as C.NamedOperator)) {
+    // return a basic named expression
+    const op = expr.name.value as any as C.NamedOperator;
+    const func = namedFunctions.get(op);
+    return new E.Named(expr.name, args, func.apply);
+  } else if (creatorConfig.type === 'sync') {
+    // Expression might be extension function, check this for the sync or async case
```

Comment:
Comment incorrect?
We're in the sync case?
(I'd add a similar comment in the async case)

Reply:
I thought that might've been overkill, I'll add another.